src/theia/sfm/global_pose_estimation/robust_rotation_estimator.cc

```diff
@@ -147,40 +146,22 @@ void RobustRotationEstimator::SetupLinearSystem() {
   sparse_matrix_.setFromTriplets(triplet_list.begin(), triplet_list.end());
 }
 
-// Computes the relative rotation error based on the current global
-// orientation estimates.
-void RobustRotationEstimator::ComputeRotationError() {
-  int rotation_error_index = 0;
-  for (const auto& relative_rotation : relative_rotations_) {
-    const Eigen::Vector3d& relative_rotation_aa = relative_rotation.second;
-    const Eigen::Vector3d& rotation1 =
-        FindOrDie(*global_orientations_, relative_rotation.first.first);
-    const Eigen::Vector3d& rotation2 =
-        FindOrDie(*global_orientations_, relative_rotation.first.second);
-
-    // Compute the relative rotation error as:
-    //   R_err = R2^t * R_12 * R1.
-    relative_rotation_error_.segment<3>(3 * rotation_error_index) =
-        MultiplyRotations(-rotation2,
-                          MultiplyRotations(relative_rotation_aa, rotation1));
-    ++rotation_error_index;
-  }
-}
-
 bool RobustRotationEstimator::SolveL1Regression() {
-  static const double kConvergenceThreshold = 1e-3;
-
   L1Solver<Eigen::SparseMatrix<double> >::Options options;
   options.max_num_iterations = 5;
   L1Solver<Eigen::SparseMatrix<double> > l1_solver(options, sparse_matrix_);
 
-  rotation_change_.setZero();
+  tangent_space_step_.setZero();
+  ComputeResiduals();
```

Comment:
I see -- so this is fixing a lag in the termination criteria where the residuals were previously not updated before the termination check. Is that correct?

Reply:
Yes, that's it. When I changed the convergence criteria I needed to fix the lag to have up-to-date step sizes when I checked if we were done.